tests/components/meteo_france/test_config_flow.py

```diff
@@ -1,29 +1,82 @@
 """Tests for the Meteo-France config flow."""
-from meteofrance.client import meteofranceError
+from meteofrance.model import Place
 import pytest
 
 from homeassistant import data_entry_flow
-from homeassistant.components.meteo_france.const import CONF_CITY, DOMAIN
+from homeassistant.components.meteo_france.const import (
+    CONF_CITY,
+    DOMAIN,
+    FORECAST_MODE_DAILY,
+    FORECAST_MODE_HOURLY,
+)
 from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_USER
+from homeassistant.const import CONF_LATITUDE, CONF_LONGITUDE, CONF_MODE
+from homeassistant.helpers.typing import HomeAssistantType
 
 from tests.async_mock import patch
 from tests.common import MockConfigEntry
 
 CITY_1_POSTAL = "74220"
 CITY_1_NAME = "La Clusaz"
-CITY_2_POSTAL_DISTRICT_1 = "69001"
-CITY_2_POSTAL_DISTRICT_4 = "69004"
-CITY_2_NAME = "Lyon"
-
-
-@pytest.fixture(name="client_1")
-def mock_controller_client_1():
+CITY_1_LAT = 45.90417
+CITY_1_LON = 6.42306
+CITY_1_COUNTRY = "FR"
+CITY_1_ADMIN = "Rhône-Alpes"
+CITY_1_ADMIN2 = "74"
+CITY_1 = Place(
+    {
+        "name": CITY_1_NAME,
+        "lat": CITY_1_LAT,
+        "lon": CITY_1_LON,
+        "country": CITY_1_COUNTRY,
+        "admin": CITY_1_ADMIN,
+        "admin2": CITY_1_ADMIN2,
+    }
+)
+
+CITY_2_NAME = "Auch"
+CITY_2_LAT = 43.64528
+CITY_2_LON = 0.58861
+CITY_2_COUNTRY = "FR"
+CITY_2_ADMIN = "Midi-Pyrénées"
+CITY_2_ADMIN2 = "32"
+CITY_2 = Place(
+    {
+        "name": CITY_2_NAME,
+        "lat": CITY_2_LAT,
+        "lon": CITY_2_LON,
+        "country": CITY_2_COUNTRY,
+        "admin": CITY_2_ADMIN,
+        "admin2": CITY_2_ADMIN2,
+    }
+)
+
+CITY_3_NAME = "Auchel"
+CITY_3_LAT = 50.50833
+CITY_3_LON = 2.47361
+CITY_3_COUNTRY = "FR"
+CITY_3_ADMIN = "Nord-Pas-de-Calais"
+CITY_3_ADMIN2 = "62"
+CITY_3 = Place(
+    {
+        "name": CITY_3_NAME,
+        "lat": CITY_3_LAT,
+        "lon": CITY_3_LON,
+        "country": CITY_3_COUNTRY,
+        "admin": CITY_3_ADMIN,
+        "admin2": CITY_3_ADMIN2,
+    }
+)
+
+
+@pytest.fixture(name="client_single")
```

Comment:
Add a client_empty to test when the search returns no cities

Reply:
Test added.